kinto2xml/exporter.py

```diff
@@ -83,32 +115,89 @@ def write_plugin_items(xml_tree, records):
     </pluginItem>
     """
 
+    if not records:
+        return
+
     pluginItems = etree.SubElement(xml_tree, 'pluginItems')
     for item in records:
-        if item.get('enabled', True):
-            entry = etree.SubElement(pluginItems, 'pluginItem',
-                                     blockID=item.get('blockID', item['id']))
-            if 'matchName' in item:
-                etree.SubElement(entry, 'match',
-                                 name='name',
-                                 exp=item['matchName'])
-            if 'matchFilename' in item:
-                etree.SubElement(entry, 'match',
-                                 name='filename',
-                                 exp=item['matchFilename'])
-            if 'matchDescription' in item:
-                etree.SubElement(entry, 'match',
-                                 name='description',
-                                 exp=item['matchDescription'])
-
-            if 'infoURL' in item:
-                infoURL = etree.SubElement(entry, 'infoURL')
-                infoURL.text = item['infoURL']
-
-            build_version_range(entry, item)
-
-
-def write_gfx_items(xml_tree, records):
+        if item.get('enabled', True) and is_related_to(item, app_id):
+            for versionRange in item.get('versionRange', []):
+                if not versionRange.get('targetApplication'):
+                    add_plugin_item(pluginItems, item, versionRange)
+                else:
+                    for targetApplication in versionRange['targetApplication']:
+                        if 'guid' not in targetApplication or \
+                           targetApplication['guid'] == app_id:
```

Comment:
the condition could be rewritten, at least with intermediary variable.

or maybe have a tiny helper for this (it looks the same as above in build_version_range)